### ISSUE TITLE
ibm_db installation using npm throws "msbuild not recognized error.." while it worked initially 2-3 weeks back without any issues

### ISSUE THREAD

**Issue by MohanrajDurai** [ibmdb/node-ibm_db#695]
Few weeks back when I started using ibm_db version 2.4.0 it worked without any issue. After I have created a new branch and tried installation, faced the **msbuild not recognized as internal or external command error**. I thought it to be a new branch issue and tried removing and reinstalling ibm_db in the old branch. Faced the same issue again which confirmed it is not with respect to branch change. 

One thing I have done is update my VS Code to version 1.48.2. I am not sure it is causing the problem as I tried digging the msbuild issue. I have tried installing the dependencies mentioned for ibm_db and nothing worked out. I am attaching my package.json and the VS Code version details below. Could you please help me on this?

**VS Code**
Version: 1.48.2 (system setup)
Commit: a0479759d6e9ea56afa657e454193f72aef85bd0
Date: 2020-08-25T10:13:11.295Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18362

**Package.json**
{
    "name": "automation",
    "version": "0.0.1",
    "license": "MIT",
    "description": "This package json includes the dev dependencies and script names to run Automation tests",
    "scripts": {
            },
    "private": true,
    "dependencies": {
        "exceljs": "^1.15.0",
        "fs-extra": "^3.0.1",
        "ibm_db": "^2.7.1",
        "jasmine-data-provider": "^2.2.0",
        "jasmine-reporters": "^2.3.1",
        "log4js": "^2.5.3",
        "log4js-protractor-appender": "^1.1.2",
        "msnodesqlv8": "^0.6.11",
        "mssql": "^4.3.0",
        "node-powershell": "^3.3.1",
        "proper-lockfile": "^3.2.0",
        "protractor-beautiful-reporter": "^1.2.1",
        "protractor-fail-fast": "^3.1.0",
        "protractor-flake": "^4.0.0",
        "protractor-pretty-html-reporter": "0.0.16",
        "request": "^2.88.2",
        "selenium-webdriver": "^4.0.0-alpha.1",
        "string-extended": "0.0.8",
        "ts-helpers": "1.1.1",
        "typescript-string-operations": "^1.3.1",
        "xlsx": "^0.12.6",
        "xml-query": "^1.5.0",
        "xml-reader": "^2.4.3"
    },
    "devDependencies": {
        "@angular/core": "^6.1.10",
        "@angular/common": "^6.1.10",
        "@angular/compiler": "^6.1.10",
        "@angular/compiler-cli": "^6.1.10",
        "@types/node": "~6.0.73",
        "@types/jasmine": "2.5.38",
        "gulp-tslint": "^8.1.2",
        "rxjs": "^6.5.3",
        "rxjs-compat": "^6.5.3",
        "zone.js": "~0.8.29",
        "jasmine-core": "^2.6.4",
        "jasmine-spec-reporter": "~4.1.0",
        "protractor": "^5.4.2",
        "ts-node": "~7.0.0",
        "tslint": "^5.20.1",
        "tslint-html-report": "^1.0.7",
        "typescript": "3.4.5"
    }
}

<img width="603" alt="ibm_db install error" src="https://user-images.githubusercontent.com/71088743/92764617-3230ab00-f3b2-11ea-85a4-b77923f18af4.png">

**Comment by bimalkjha** [comment]
@MohanrajDurai Close vscode. Open windows command prompt and cd to ibm_db directory as mentioned in first line of error. Then execute command `npm install` under ibm_db directory. It should print message success near end.
Open vscode and try. It should work. If it did not work, share the complete output of `npm install` command. Thanks.

**Comment by bimalkjha** [comment]
@MohanrajDurai If it still do not work. Then cd to db2connect extension directory and try below commands:
```
npm uninstall ibm_db
npm install ibm_db
cd node_modules/ibm_db
npm install
```
In next release of vscode extension, we'll get an option to do it when vscode version get updated and no need to run these commands from command line. Thanks.

**Comment by MohanrajDurai** [comment]
@bimalkjha I have tried your first suggestion and it is not helping. Does that mean i
[Try 1.txt](https://github.com/ibmdb/node-ibm_db/files/5218421/Try.1.txt)
[installation.txt](https://github.com/ibmdb/node-ibm_db/files/5218422/installation.txt)

nstalling all my packages in the ibm_db directory? I am attaching the tried console log and npm installation console log. 

Also I couldn't understand the second suggestion as we dont have DB2connect extension available in our package

**Comment by bimalkjha** [comment]
@MohanrajDurai Error in `1.txt` is expected as `ibm_db` was in use and your `vscode` window was open.
`installation.txt` looks good. You are hitting error from msbuild as below and you need to set some PATH for it:
```
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Users\jisamd2\Documents\OM_Auto_Master_0831\app_ordermanagement_order_automation\node_modules\ibm_db\build\o
dbc_bindings.vcxproj(21,3): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that 
the path in the <Import> declaration is correct, and that the file exists on disk.`
```
One thing is missing is `-vscode` option with `npm install`. Since, you want to use `ibm_db` for vscode environment you need to use `npm install -vscode` command under `ibm_db` directory. It is documented [here](https://github.com/ibmdb/node-ibm_db#usage-within-vs-code)
The `npm install -vscode` command should work without any `windows-build-tools`. To know the installation of `ibm_db` is happening correctly for vscode is presence of a line `Proceeding to build IBM_DB for Electron framework...` in install output.
Thanks.

**Comment by bimalkjha** [comment]
If you want to compile yourself using `msbuild` after correcting the PATH, make sure you have installed the same version of node.js in your system as shown by the `Help->About` option of VSCode. If you are still hitting issue, please share the complete output of `npm install -vscode` command and output of `Help->About` option of vscode. Thanks.

**Comment by MohanrajDurai** [comment]
Hi @bimalkjha , We have analyzed different versions and found In 2.7.1:

https://github.com/ibmdb/node-ibm_db/blob/v2.7.1/lib/odbc.js
![MicrosoftTeams-image (12)](https://user-images.githubusercontent.com/71088743/94232482-babb5980-ff23-11ea-965c-87ce8f9d41a7.png)

Highlighted part of code is causing issue for us in 2.7.1, if we append CLI Driver after PATH like below then its working - Not sure why.
 
  process.env.PATH = process.env.PATH + ';' + path.resolve(clidriver, 'bin') + ';' +
                      path.resolve(clidriver, 'bin/amd64.VC12.CRT') + ';' +
                      path.resolve(clidriver, 'bin/icc64') + ';' +
                      path.resolve(clidriver, 'lib');

So we have switched the version to version 2.7.0 which helped resolve the License error when we try to connect to DB2(which we have thought it could be due top msbuild error. attached the screenshot below). Even now the installation is throwing error. But we are able to connect to DB2. 
<img width="960" alt="ibm_db run error" src="https://user-images.githubusercontent.com/71088743/94232851-7ed4c400-ff24-11ea-9794-633ed6359ac3.png">


Thanks for you help !!!
